core/src/main/java/org/mvcspec/ozark/core/ViewResponseFilter.java

```diff
@@ -147,6 +147,12 @@ public void filter(ContainerRequestContext requestContext,
                 throw new ServerErrorException(messages.get("EntityToStringNull", resourceInfo.getResourceMethod()), INTERNAL_SERVER_ERROR);
             }
             responseContext.setEntity(new Viewable(view), null, responseContext.getMediaType());
+        } else if (entityType == Viewable.class) {
```

Comment:
This will definitely work. But I think we can make the code a bit simpler. Basically this `if` block is almost the same as the `if` block above. The only difference is, that this block gets the view name from `Viewable.getView()` and the block above by calling `entity.toString()`. The rest of the code is exactly the same. 

So may we can merge these blocks into a single `else` block which gets the view name like this:

```java
final String view = appendExtensionIfRequired(
  entityType == Viewable.class ? ((Viewable) entity).getView() : entity.toString()
);
```

This class is getting more and more complex to reason about, so simplifying it is a good thing.

Do you think this can work? Or am I missing something?

Reply:
This way you put it was much simpler, but the class is complex.